.github/workflows/preview.yml

```diff
@@ -0,0 +1,37 @@
+# .github/workflows/preview.yml
+name: Deploy PR previews
+concurrency: preview-${{ github.ref }}
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - closed
+jobs:
+  deploy-preview:
+    env:
+      NEXT_PUBLIC_FIREBASE_API_KEY: ${{ secrets.NEXT_PUBLIC_FIREBASE_API_KEY }}
+      NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN: ${{ secrets.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN }}
+      NEXT_PUBLIC_FIREBASE_PROJECT_ID: ${{ secrets.NEXT_PUBLIC_FIREBASE_PROJECT_ID }}
+      NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET: ${{ secrets.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET }}
+      NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID: ${{ secrets.NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID }}
+      NEXT_PUBLIC_FIREBASE_APP_ID: ${{ secrets.NEXT_PUBLIC_FIREBASE_APP_ID }}
+      NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID: ${{ secrets.NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Clean build output
+        if: github.event.action != 'closed'
+        run: rm -rf .next out
+      - name: Set PR base path
+        if: github.event.action != 'closed'
+        run: echo "NEXT_PUBLIC_PR_BASE_PATH=/pr-preview/pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+      - run: npm i && npm run build
+        if: github.event.action != 'closed'
+      - uses: rossjrw/pr-preview-action@v1
```

Comment:
can refactor this.

something like
```

    steps:
      - uses: actions/checkout@v4

      - name: Prepare Build Environment
        if: github.event.action != 'closed'
        run: |
          rm -rf .next out
          echo "NEXT_PUBLIC_PR_BASE_PATH=/pr-preview/pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV

      - name: Install Dependencies & Build
        if: github.event.action != 'closed'
        run: npm i && npm run build

      - uses: rossjrw/pr-preview-action@v1
```


havent tested if that will work or not

Reply:
Cleaner, I like it. We'll find out if it works depending on whether or not the PR preview goes down in ~5 min 🤣

Btw we won't know if the `deploy.yml` chart will need some TLC or not until this change goes in, due to it needing changes both in the code & the repo settings